src/components/Sidebar/ActionNode.tsx

```diff
@@ -1,49 +1,212 @@
+      action.resource.code.coding[0].code = code;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
+
+  const changeTitle = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const title = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
+      action.resource.title = title;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
 
+  const selectActionType = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+      const value = event?.target.value || '';
+      const actionType = actionTypeOptions.find(option => {
+        return option.value === value;
+      });
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      let action: Action;
+      if (actionType?.label === 'Careplan') {
+        action = {
+          type: 'create',
+          description: '',
+          resource: {
+            id: shortid.generate(),
+            resourceType: actionType?.label,
+            title: ''
+          }
+        };
+      } else if (
+        currentState.action.length > 0 &&
+        currentState.action[0].resource.resourceType !== 'Careplan'
+      ) {
+        action = currentState.action[0];
+        action.resource.resourceType = actionType?.label;
+      } else {
+        action = {
+          type: 'create',
+          description: '',
+          resource: {
+            id: shortid.generate(),
+            resourceType: actionType?.label,
+            code: {
+              coding: [
+                {
+                  system: '',
+                  code: '',
+                  display: ''
+                }
+              ]
+            }
+          }
+        };
+      }
+
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
+
+  const selectCodeSystem = useCallback(
+    (event: ChangeEvent<{ value: string }>): void => {
+      if (!currentNodeKey) return;
+
+      const codeSystem = event?.target.value || '';
+      const currentState: GuidanceState = pathway.states[currentNodeKey] as GuidanceState;
+      const action: Action = currentState.action[0];
+      action.resource.code.coding[0].system = codeSystem;
+      updatePathway(setStateAction(pathway, currentNodeKey, [action]));
+    },
+    [currentNodeKey, pathway, updatePathway]
+  );
   // The node has a key and is not the start node
   const changeNodeTypeEnabled = currentNode.key !== undefined && currentNode.key !== 'Start';
 
+  const action = currentNode.action;
+  const resource = action && action.length > 0 && action[0].resource;
```

Comment:
The null check for action can be added to the second statement
```suggestion
  const resource = action?.length > 0 && action[0].resource;
```